dtfs-central-api/src/repositories/tfm-deals-repo/tfm-deal-cancellation.repo.submitScheduledDealCancellation.test.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  DEAL_SUBMISSION_TYPE,
+  DealNotFoundError,
+  InvalidDealIdError,
+  MONGO_DB_COLLECTIONS,
+  TFM_DEAL_CANCELLATION_STATUS,
+  TFM_DEAL_STAGE,
+  TfmActivity,
+} from '@ukef/dtfs2-common';
+import { generateAuditDatabaseRecordFromAuditDetails, generateTfmAuditDetails } from '@ukef/dtfs2-common/change-stream';
+import { ObjectId } from 'mongodb';
+import { getUnixTime } from 'date-fns';
+import { mongoDbClient as db } from '../../drivers/db-client';
+import { TfmDealCancellationRepo } from './tfm-deal-cancellation.repo';
+import { aTfmUser } from '../../../test-helpers';
+
+const dealId = new ObjectId();
+
+const mockBankRequestDate = getUnixTime(new Date('2024-01-01'));
+const mockEffectiveFrom = getUnixTime(new Date('2025-02-02'));
+const mockReason = 'mock reason';
+
+const mockDealCancellationObject = { reason: mockReason, bankRequestDate: mockBankRequestDate, effectiveFrom: mockEffectiveFrom };
+
+const tfmUserId = aTfmUser()._id;
+
+const auditDetails = generateTfmAuditDetails(tfmUserId);
+
+const mockActivity = { text: 'This is an activity' } as TfmActivity;
+
+describe('tfm-deals-cancellation-repo', () => {
+  const updateOneMock = jest.fn();
+  const updateManyMock = jest.fn();
+  const findMock = jest.fn();
+  const findToArrayMock = jest.fn();
+  const getCollectionMock = jest.fn();
+
+  const matchingFacilityId1 = new ObjectId();
+  const matchingFacilityId2 = new ObjectId();
+
+  const mockMatchedFacilities = [{ facilitySnapshot: { ukefFacilityId: matchingFacilityId1 } }, { facilitySnapshot: { ukefFacilityId: matchingFacilityId2 } }];
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('submitScheduledDealCancellation', () => {
+    const mockUpdateResult = { matchedCount: 1 };
+
+    beforeAll(() => {
+      jest.useFakeTimers();
+    });
+
+    beforeEach(() => {
+      updateOneMock.mockResolvedValue(mockUpdateResult);
+      findToArrayMock.mockResolvedValue(mockMatchedFacilities);
+      findMock.mockReturnValue({ toArray: findToArrayMock });
+
+      getCollectionMock.mockResolvedValue({
+        updateOne: updateOneMock,
+        updateMany: updateManyMock,
+        find: findMock,
+      });
+      jest.spyOn(db, 'getCollection').mockImplementation(getCollectionMock);
```

Comment:
since we're calling two different collections we could use

`when(getCollectionMock).calledWith('facilities').mockImplementation()`

Reply:
Good spot, have amended thanks!